.github/actions/setup-job-macos/action.yml

```diff
@@ -0,0 +1,94 @@
+name: 'Setup job for self-hosted macOS runner'
+description: 'Setup rust and flutter'
+inputs:
+  flutter:
+    description: 'Setup flutter'
+    default: "false"
+    required: false
+  rust-target:
+    description: 'Install an additional rust target'
+    default: ""
+    required: false
+  cache-key:
+    description: 'An optional key that is added to the automatic cache key'
+    default: ""
+    required: false
+  github_token:
+    description: 'The github token used for the `just` action to avoid exceeding API rate limits'
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: Clean up runner tool cache and temp dir
+      shell: bash
+      run: |
+        cd $RUNNER_TOOL_CACHE
+        rm -rf *
+        cd $RUNNER_TEMP
+        rm -rf *
+
+    - name: Load .env
+      uses: xom9ikk/dotenv@d3ff95524814ceac377510f30f4af6296ea612c1 #v1.0.2
+
+    - name: Install just
+      uses: extractions/setup-just@aa5d15c144db4585980a44ebfdd2cf337c4f14cb #v1.4.0
+      env:
+        GITHUB_TOKEN: ${{ inputs.github_token }}
+      with:
+        just-version: ${{ env.JUST_VERSION }}
+
+    - name: Install flutter
+      if: inputs.flutter == 'true'
+      uses: subosito/flutter-action@5e1529bc12adae5b26f30f0a4046ac743d6f0c5d # v2.3.0
+      with:
+        flutter-version: ${{ env.FLUTTER_VERSION }}
+
+    - name: Setup flutter dependencies
+      if: inputs.flutter == 'true'
+      shell: bash
+      run: just flutter-deps
+
+    - if: inputs.flutter == 'true'
+      shell: bash
+      run: |
+        echo "GEM_HOME=$RUNNER_TOOL_CACHE/.gem" >> $GITHUB_ENV
```

Comment:
🎨 I would merge this with the other path-setting code below. We don't gain anything, runtime wise, by keeping them apart / conditional, and it'll provide a better overview of the environment (imo).

Reply:
I prefer to keep them separate because I think it's easier to understand which tool belongs to which toolchain